aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/UpdateHandler.java

```diff
@@ -119,4 +139,95 @@ private boolean shouldRemoveFromGlobalCluster(
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final ResourceModel resourceModel
+    ) {
+        DBCluster cluster;
+        try {
+            cluster = fetchDBCluster(proxyClient, resourceModel);
+        } catch (Exception ex) {
+            return Commons.handleException(progress, ex, DEFAULT_DB_CLUSTER_ERROR_RULE_SET);
+        }
+
+        ProgressEvent<ResourceModel, CallbackContext> result = progress;
+
+        for (final DBClusterMember dbClusterMember: cluster.dbClusterMembers()) {
+            final ProgressEvent<ResourceModel, CallbackContext> p = proxy.initiate("rds::reboot-dbcluster-member-" + dbClusterMember.dbInstanceIdentifier(),
+                            proxyClient,
+                            progress.getResourceModel(),
+                            progress.getCallbackContext())
+                    .translateToServiceRequest(model -> Translator.rebootDbInstanceRequest(dbClusterMember.dbInstanceIdentifier()))
+                    .backoffDelay(config.getBackoff())
+                    .makeServiceCall((rebootDbInstanceRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(
+                            rebootDbInstanceRequest,
+                            proxyInvocation.client()::rebootDBInstance
+                    ))
+                    .stabilize((rebootDbInstanceRequest, rebootDbInstanceResponse, proxyInvocation, model, context) ->
+                            isDBInstanceStabilized(proxyClient, dbClusterMember.dbInstanceIdentifier(), model))
+                    .handleError((createRequest, exception, client, model, callbackCtx) -> Commons.handleException(
+                            ProgressEvent.progress(model, callbackCtx),
+                            exception,
+                            DEFAULT_DB_CLUSTER_ERROR_RULE_SET
+                    ))
+                    .progress();
+
+            if (p.isFailed() || p.isInProgressCallbackDelay()) {
+                return p;
+            }
+            result = p;
+        }
+        return result;
+    }
+
+    private boolean shouldRebootCluster(final ResourceModel previousResourceState, final ResourceModel desiredResourceState, boolean isRollback) {
+        return !isRollback &&
+                !Objects.equals(previousResourceState.getEngineVersion(), desiredResourceState.getEngineVersion()) &&
+                !Objects.equals(previousResourceState.getDBInstanceParameterGroupName(), desiredResourceState.getDBInstanceParameterGroupName());
+    }
+
+    private boolean isDBInstanceStabilized(final ProxyClient<RdsClient> client, final String dbInstanceIdentifier, final ResourceModel resourceModel) {
+        final DBInstance dbInstance = fetchDBInstance(client, dbInstanceIdentifier);
+        final DBCluster dbCluster = fetchDBCluster(client, resourceModel);
+
+        return "available".equals(dbInstance.dbInstanceStatus()) &&
```

Comment:
I know 2 more attributes that needed to be checked as well in DBInstance in case of update `DBInstanceClass, EngineVersion` as we saw failures before due to limited infrastructure capacity in region which result to modification is not applied although the API call succeeded and DBInstance returned to available status again.

Reply:
Hence, like in one of the comments I think that this kind of stabilization logic should be moved to a common place. But I belive that would be better done in a separate PR.